### DISCUSSION TITLE
Local Development - Secrets management for OAuth

### DISCUSSION THREAD

**Discussion by samwoolertonFosters** [supabase/supabase#8499] [Questions]
How do you manage secrets locally when configuring OAuth connections? I'm configuring Azure and expect that committing `client_id` and `secret` to Git is a bad idea.
Doesn't look like `.toml` files can include/import each other, so is there a way to read from environment variables?

Open to options, just looking to keep secrets out of Git where possible

**Comment by J0** [comment]
Hey @samwoolertonFosters, 

Not sure I fully got the question so do feel free to correct me if I'm misunderstanding the question. 

For reading from environment variables one can typically use  something like [process.env](https://codeburst.io/process-env-what-it-is-and-why-when-how-to-use-it-effectively-505d0b2831e7) for node or `dotenv` in python or similar. 

Do you mind sharing more details about your use case?

**Reply by samwoolertonFosters** [reply]
Sorry if it wasn't clear, I meant configuring Supabase's auth (per [the docs](https://supabase.com/docs/guides/auth)).
Configuring via the UI works fine in production but when running Supabase locally I want to keep secrets out of Git instead of adding to the Supabase `config.toml`.

How do most people run Supabase locally for dev purposes if they're using OAuth providers?

**Reply by sroussey** [reply]
I ran into this as well. Seems like the best solution would be to do like .env files and have .local read in after and merge the two.

**Reply by sroussey** [reply]
Looking at https://github.com/supabase/cli/blob/cd1481b299deaba161a332091cfb0dec7b5abe6f/internal/utils/config.go#L164 I do not see anything that they have added to help this situation. toml files do not natively use env expansion, so the two files (one git ignored) would be the best bet. Particularly since most providers don't have test/prod secrets to use

**Reply by fletchertyler914** [reply]
I just ran into this issue too. I am using github to login, but don't want to generate a new key or delete->commit->copy->paste from the env variables manually every time I want to push a commit. 

I could add the config to the .gitignore now that I pushed the base settings, but then any other updates I want to push would require the same song and dance to ensure I don't commit the secret. 

There should be a way for that to be updated from the env on docker build or something.
